specification/syntex/resource-manager/Microsoft.Syntex/preview/2022-09-15-preview/syntex.json

```diff
@@ -0,0 +1,521 @@
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ProcessorNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "body",
+            "description": "Document processor details.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DocumentProcessorPatchableProperties"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DocumentProcessorResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "DocumentProcessor_Update": {
+            "$ref": "./examples/DocumentProcessor_Update.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Syntex/operations": {
+      "get": {
+        "operationId": "Operations_List",
+        "description": "List of all REST operations supported by the Microsoft.Syntex provider.",
+        "tags": [
+          "Operations"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Describes the result of a successful operation.",
+            "schema": {
+              "$ref": "#/definitions/OperationsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DocumentProcessorResource": {
+      "type": "object",
+      "description": "Document processor details",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "readOnly": true
+        },
+        "properties": {
+          "x-ms-client-flatten": false,
+          "description": "Document processor properties.",
+          "$ref": "#/definitions/DocumentProcessorProperties"
+        }
+      }
+    },
+    "DocumentProcessorPatchableProperties": {
```

Comment:
>DocumentProcessorPatchableProperties [](http://example.com/codeflow?start=5&length=36)

Can possibly be merged with DocumentProcessorProperties.

Reply:
They cannot be merged because DocumentProcessorProperties are required and DocumentProcessorPatchableProperties are optional - we need to be able to update one property without affecting the other.  RPaaS validation would not allow updating a single property if DocumentProcessorProperties were used.

Reply:
I think there might also be a way around that, like only adding the required tags on the DocumentProcessorProperties, but it doesn't really matter.